modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/freelist/FreeListImpl.java

```diff
@@ -84,300 +86,59 @@ public abstract class AbstractFreeList<T extends Storable> extends PagesList imp
-
-                    if (io.getFreeSpace(pageAddr) < row.size() - written) {
-                        break;
-                    }
-                }
-
-                written = writeRowHnd.addRow(pageId, pageAddr, io, row, written);
-
-                assert written == COMPLETE;
-
-                evictionTracker.touchPage(pageId);
-            }
-
-            writeRowHnd.putPage(io.getFreeSpace(pageAddr), pageId, pageAddr, statHolder);
-
-            return written;
-        }
-    }
-
-    private final class RemoveRowHandler implements PageHandler<ReuseBag, Long> {
-        /** Indicates whether partition ID should be masked from page ID. */
-        private final boolean maskPartId;
-
-        /**
-         * Constructor.
-         *
-         * @param maskPartId Indicates whether partition ID should be masked from page ID.
-         */
-        RemoveRowHandler(boolean maskPartId) {
-            this.maskPartId = maskPartId;
-        }
-
-        /** {@inheritDoc} */
-        @Override
-        public Long run(
-                int cacheId,
-                long pageId,
-                long page,
-                long pageAddr,
-                PageIo iox,
-                ReuseBag reuseBag,
-                int itemId,
-                IoStatisticsHolder statHolder
-        ) throws IgniteInternalCheckedException {
-            AbstractDataPageIo<T> io = (AbstractDataPageIo<T>) iox;
-
-            int oldFreeSpace = io.getFreeSpace(pageAddr);
-
-            assert oldFreeSpace >= 0 : oldFreeSpace;
-
-            long nextLink = io.removeRow(pageAddr, itemId, pageSize());
-
-            int newFreeSpace = io.getFreeSpace(pageAddr);
-
-            if (newFreeSpace > MIN_PAGE_FREE_SPACE) {
-                int newBucket = bucket(newFreeSpace, false);
-
-                boolean putIsNeeded = oldFreeSpace <= MIN_PAGE_FREE_SPACE;
-
-                if (!putIsNeeded) {
-                    int oldBucket = bucket(oldFreeSpace, false);
-
-                    if (oldBucket != newBucket) {
-                        // It is possible that page was concurrently taken for put, in this case put will handle bucket change.
-                        pageId = maskPartId ? PageIdUtils.maskPartitionId(pageId) : pageId;
-
-                        putIsNeeded = removeDataPage(pageId, pageAddr, io, oldBucket, statHolder);
-                    }
-                }
-
-                if (io.isEmpty(pageAddr)) {
-                    evictionTracker.forgetPage(pageId);
-
-                    if (putIsNeeded) {
-                        reuseBag.addFreePage(recyclePage(pageId, pageAddr));
-                    }
-                } else if (putIsNeeded) {
-                    put(null, pageId, pageAddr, newBucket, statHolder);
-                }
-            }
-
-            // For common case boxed 0L will be cached inside of Long, so no garbage will be produced.
-            return nextLink;
-        }
-    }
+    private final IoStatisticsHolder statHolder;
```

Comment:
When I asked about static classes, I didn't mean moving them into other files. And the goal was to make fields static as well. If fields are not static then what's the point? I think I should have expressed myself more clear last time.
Also, please ask me if you're not sure what I'm expecting. What you did right now it pretty much useless and it bloats PR for no good reason :(